app/views/components/_modal_dialogue.html.erb

```diff
@@ -0,0 +1,31 @@
+<%
+  id ||= "modal-dialogue-#{SecureRandom.hex(4)}"
+  heading_level ||= 2
+  title ||= nil
+  wide ||= false
+%>
+<button class="govuk-button" data-toggle="modal" data-target="#<%= id %>">Launch modal dialogue</button>
+
+<div class="app-c-modal-dialogue" data-module="modal-dialogue" id="<%= id %>">
+  <dialog class="app-c-modal-dialogue__box <%= 'app-c-modal-dialogue__box--wide' if wide %>"
```

Comment:
Is using the `dialog` element okay? Seems like it's [not widely supported](https://caniuse.com/#search=dialog) although it seems to work okay in most of the browsers I've tested, oddly (although it doesn't seem to work at all in IE10, but I don't know what's the cause).

Reply:
Not widely supported, but follows the standard, is semantically correct and provides more context for screen readers in Chrome and Opera. Tested in IE10 and seems to be working fine. What problem did you encounter?

Reply:
I'm using IE10 in browserstack and the component guide doesn't do anything when clicking on the button to bring up the modal. Does that work for you or are you testing in a different context?

Reply:
Yes, same setup here (both win 7 and 8)
![screen shot 2019-02-01 at 14 00 40](https://user-images.githubusercontent.com/788096/52127386-d88acb00-2629-11e9-8e0a-dceafec1cd47.png)